awx/api/serializers.py

```diff
@@ -995,6 +995,24 @@ def validate_password(self, value):
         django_validate_password(value)
         if not self.instance and value in (None, ''):
             raise serializers.ValidationError(_('Password required for new User.'))
+
+        # Check if a password is too long
+        password_max_length = User._meta.get_field('password').max_length
+        if len(value) > password_max_length:
+            raise serializers.ValidationError(_('Password max length is {}'.format(password_max_length)))
+        if getattr(settings, 'LOCAL_PASSWORD_MIN_LENGTH', 0) and len(value) < getattr(settings, 'LOCAL_PASSWORD_MIN_LENGTH'):
```

Comment:
if these settings are in the db with default values already, would
```python3
if len(value) < getattr(settings, 'LOCAL_PASSWORD_MIN_LENGTH'):
```
suffice?

Reply:
Probably, I was thinking that since 0 means "no check" we would start with that to prevent the second checks from even being looked at. But I suppose its only a string compare/small loops so the amount of time saved by the short circuiting the if clause is probably negligible.